web/src/main/java/com/epam/esm/controller/hateoas/HateoasModel.java

```diff
@@ -0,0 +1,36 @@
+package com.epam.esm.controller.hateoas;
+
+import com.epam.esm.dto.IdentifiableDto;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.RepresentationModel;
+
+import java.util.List;
+
+@Data
+@AllArgsConstructor
+@EqualsAndHashCode(callSuper = false)
+public class HateoasModel extends RepresentationModel<HateoasModel> {
+    private Object data;
+
+    private HateoasModel() {
+
+    }
+
+    public static <T extends IdentifiableDto> HateoasModel build(HateoasProvider<T> hateoasProvider, T object) {
+        HateoasModel hateoasModel = new HateoasModel(object);
+        List<Link> links = hateoasProvider.provide(object);
+        hateoasModel.add(links);
+
+        return hateoasModel;
+    }
+
+    public static <T extends IdentifiableDto> List<HateoasModel> build(HateoasProvider<T> hateoasProvider,
```

Comment:
Could you please send me an example of a response with links for "get all certificates" end-point?

Reply:
`http://localhost:8080/api/certificates/?page=1&pageSize=1`

![image](https://user-images.githubusercontent.com/42818718/137079508-b5ace7a6-8907-4092-989b-d7eff3e1c592.png)

Reply:
Fixed.